src/Playwright/BrowserContext.cs

```diff
@@ -30,6 +31,31 @@ internal BrowserContext(IChannelOwner parent, string guid, BrowserContextInitial
             Channel.Page += Channel_OnPage;
             Channel.BindingCall += Channel_BindingCall;
             Channel.Route += Channel_Route;
+            Channel.RequestFailed += (_, e) =>
+            {
+                e.Request.Failure = e.FailureText;
+                e.Request.Timing.ResponseEnd = e.ResponseEndTiming;
+                RequestFailed?.Invoke(this, e.Request);
+                e.Page?.FireRequestFailed(e.Request);
+            };
+            Channel.Request += (_, e) =>
+            {
+                Request?.Invoke(this, e.Request);
+                e.Page?.FireRequest(e.Request);
+            };
+            Channel.RequestFinished += (_, e) =>
+            {
+                e.Request.Timing.ResponseEnd = e.ResponseEndTiming;
+                RequestFinished?.Invoke(this, e.Request);
+                e.Page?.FireRequestFinished(e.Request);
+            };
+            Channel.Response += (_, e) =>
+            {
+                Response?.Invoke(this, e.Response);
+                e.Page?.FireResponse(e.Response);
+            };
+
+            _tracing = new Tracing(Channel);
```

Comment:
```suggestion
            Tracing = new Tracing(Channel);
```

Reply:
I am following the Page._mouse pattern here.

Reply:
Good. We need to refactor that.